http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/LeonardoService.scala

```diff
@@ -116,9 +116,15 @@ case class IllegalLabelKeyException(labelKey: String)
 case class InvalidDataprocMachineConfigException(errorMsg: String)
     extends LeoException(s"${errorMsg}", StatusCodes.BadRequest)
 
-case class ImageNotFoundException(traceId: TraceId, image: ContainerImage)
+final case class ImageNotFoundException(traceId: TraceId, image: ContainerImage)
     extends LeoException(s"${traceId} | Image ${image.imageUrl} not found", StatusCodes.NotFound)
 
+final case class InvalidImage(traceId: TraceId, image: ContainerImage)
+    extends LeoException(
+      s"${traceId} | Image ${image.imageUrl} doesn't have neither JUPYTER_HOME nor RSTUDIO_HOME defined. Make sure your custom image extends from one of the terra base images",
```

Comment:
Just a wording suggestion:

"Image ${image.imageUrl} doesn't have JUPYTER_HOME or RSTUDIO_HOME environment variables defined. Make sure your custom image extends from one of the Terra base images."